core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java

```diff
@@ -94,7 +96,9 @@ public AbstractEventContext(FlowExceptionHandler exceptionHandler, int depthLeve
                               Optional<CompletableFuture<Void>> externalCompletion) {
     this.depthLevel = depthLevel;
     this.externalCompletion = externalCompletion.orElse(null);
-    externalCompletion.ifPresent(completableFuture -> completableFuture.thenAccept((aVoid) -> tryTerminate()));
+    if (this.externalCompletion != null) {
+      this.externalCompletion.thenAccept((aVoid) -> tryTerminate());
+    }
```

Comment:
why this change? does this have a perf impact?

Reply:
* easier to put breakpoints
*  in theory it should reduce GC impact by avoiding a lambda that's not adding much value. 
* seems to me like abusing the Optional API, which is often misused as a replacement for an `if`